central/networkgraph/flow/datastore/internal/store/postgres/store.go

```diff
@@ -350,13 +331,8 @@ func (s *flowStoreImpl) retryableUpsertFlows(ctx context.Context, flows []*stora
 	return s.copyFrom(ctx, lastUpdateTS, flows...)
 }
 
-func (s *flowStoreImpl) acquireConn(ctx context.Context, op ops.Op, typ string) (*postgres.Conn, func(), error) {
-	defer metrics.SetAcquireDBConnDuration(time.Now(), op, typ)
-	conn, err := s.db.Acquire(ctx)
-	if err != nil {
-		return nil, nil, err
-	}
-	return conn, conn.Release, nil
+func (s *flowStoreImpl) begin(ctx context.Context) (*postgres.Tx, context.Context, error) {
+	return postgres.NewTransactionOrFromContext(ctx, s.db)
 }
```

Comment:
This repeats and does not look that store-dependent (beyond s.db). Can it be inlined or extracted ?